### PR TITLE
fix: #2996

### PR DESCRIPTION
**Pull request by liuzhenying**
##### Checklist

- [X] `npm test` passes
- [X] tests and/or benchmarks are included
- [X] commit message follows commit guidelines

##### Description of change

解决 https://github.com/antvis/G2/issues/2996

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "lint:tslint": "tslint -c tslint.json 'src/**/*.ts'",
     "lint:md": "lint-md ./docs",
     "test": "jest",
-    "test-live": "DEBUG_MODE=1 jest --watch tests/unit",
+    "test-live": "DEBUG_MODE=1 jest --watch tests/bugs/2996-spec.ts",
```
Comment (hustcc) [inline package.json:27]: 删除。

Comment (liuzhenying) [reply package.json:27]: 啊这个居然提上来了，我后来删了，再过 pre-commit

File: src/util/scale.ts [hunk 2]
```diff
@@ -80,3 +81,35 @@ export function syncScale(scale: Scale, newScale: Scale) {
...
+    range = [0.5, 1]; // 只有一个分类时,防止计算出现 [0.5,0.5] 的状态
+  } else {
+    let widthRatio = 1;
+    let offset = 0;
+
+    if (isFullCircle(coordinate)) {
+      if (!coordinate.isTransposed) {
+        range = [0, 1 - 1 / count];
+      } else {
+        widthRatio = get(this.theme, 'widthRatio.multiplePie', 1 / 1.3);
```
Comment (hustcc) [inline src/util/scale.ts:105]: 这里的 this.theme 找不到吧。

File: src/util/scale.ts [hunk 3]
```diff
@@ -80,3 +81,35 @@ export function syncScale(scale: Scale, newScale: Scale) {
 export function getName(scale: Scale): string {
   return scale.alias || scale.field;
 }
+
+/**
+ * 根据 scale values 和 coordinate 获取分类默认 range
+ * @param scale 需要获取的 scale 实例
+ * @param coordinate coordinate 实例
+ */
+export function getDefaultCategoryScaleRange(scale: Scale, coordinate: Coordinate): Scale['range']{
```
Comment (hustcc) [inline src/util/scale.ts:90]: 最后一个括号前面要有空格。

**Review by hustcc — APPROVED** [review]

**Pull request merged**